src/EditorFeatures/Core/FindUsages/ClassifiedSpansAndHighlightSpanFactory.cs

```diff
@@ -44,6 +47,82 @@ internal static class ClassifiedSpansAndHighlightSpanFactory
                 documentSpan.Document, documentSpan.SourceSpan, cancellationToken).ConfigureAwait(false);
         }
 
+        public static async Task<ClassifiedTextElement> BuildClassifiedTextElementForSpansAsync(ImmutableArray<TextSpan> spans, ITextSnapshot snapshot, Document document, CancellationToken cancellationToken)
+        {
+            var classificationService = document.GetLanguageService<IClassificationService>();
+
+            var classifiedSpans = new List<ClassifiedSpan>();
+            var semanticSpans = new List<ClassifiedSpan>();
+
+            foreach (var span in spans)
+            {
+                await classificationService.AddSyntacticClassificationsAsync(document, span, classifiedSpans, cancellationToken).ConfigureAwait(false);
+                await classificationService.AddSemanticClassificationsAsync(document, span, semanticSpans, cancellationToken).ConfigureAwait(false);
+            }
```

Comment:
Why do we need this function when we already have: GetClassifiedSpansAsync?

Can we not just call that, then convert the results to ClassifiedTextElements?

Reply:
Thanks!